HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt

```diff
@@ -857,48 +795,33 @@ class HyprMXAdapter : PartnerAdapter {
             )
         }
 
-        override fun onAdDisplayError(
-            placement: Placement?,
-            hyprMXError: HyprMXErrors?,
-        ) {
+        override fun onAdDisplayError(placement: Placement, hyprMXError: HyprMXErrors) {
             PartnerLogController.log(SHOW_FAILED)
-            hyprMXError?.let {
+            hyprMXError.let {
                 onShowError(it)
                 onShowError = {}
             }
         }
 
-        override fun onAdAvailable(placement: Placement?) {
-            PartnerLogController.log(LOAD_SUCCEEDED)
-            resumeOnce(
-                Result.success(
-                    PartnerAd(
-                        ad = placement,
-                        details = emptyMap(),
-                        request = request,
-                    ),
-                ),
-            )
-        }
-
-        override fun onAdNotAvailable(placement: Placement?) {
-            PartnerLogController.log(LOAD_FAILED)
-            resumeOnce(
-                Result.failure(
-                    ChartboostMediationAdException(
-                        ChartboostMediationError.CM_LOAD_FAILURE_NO_FILL,
-                    ),
+        override fun onAdImpression(placement: Placement) {
+            PartnerLogController.log(DID_TRACK_IMPRESSION)
+            listener?.onPartnerAdImpression(
+                PartnerAd(
+                    ad = placement,
+                    details = emptyMap(),
+                    request = request,
                 ),
             )
         }
+    }
 
-        override fun onAdRewarded(
-            placement: Placement,
-            rewardName: String,
-            rewardValue: Int,
-        ) {
+    private class ShowRewardedAdListener(
+        private val request: PartnerAdLoadRequest,
+        private val listener: PartnerAdListener?,
+    ) : ShowAdListener(request, listener), HyprMXRewardedShowListener {
```

Comment:
This was added due to the issue https://chartboost.atlassian.net/browse/HB-7718 during certification.